comms/core/src/connectivity/manager.rs

```diff
@@ -662,6 +664,19 @@ impl ConnectivityManagerActor {
         Ok(())
     }
 
+    async fn mark_peer_last_seen_now(&self, conn: &PeerConnection) -> Result<(), ConnectivityError> {
+        // Can't be None
+        if let Some(mut peer) = self.peer_manager.find_by_node_id(conn.peer_node_id()).await? {
+            if let Some(addr) = conn.known_address() {
+                peer.addresses.mark_last_seen_now(addr);
+            } else {
+                peer.addresses.mark_all_addresses_as_last_seen_now();
```

Comment:
I would prefer us only mark the address we have and we connected to as last seen true. 
If a peer has multiple addresses with only one if them true, we only need to mark that one as the address as last seen and not all.